tensorflow/python/autograph/operators/py_builtins.py

```diff
@@ -25,6 +25,7 @@
 
 import six
 
+from tensorflow.data import Dataset
```

Comment:
Please import the internal module instead:

`from tensorflow.python.data.ops import dataset_ops`

Then the check below can be:

`if isinstance(s, dataset_ops.DatasetV2):`

Reply:
Yes, it can be changed!
I'm sorry if this is a dumb question, but why it needed to be done that way? Thank you

Reply:
Forgot to answer this one - when we import the public Dataset symbol, we're practically importing the entire TensorFlow, which creates a circular dependency chain (because TF includes autograph). Instead, if we directly import the implementation class then we avoid the circular dependency and at the same time only import the module that we needed.